assembly/src/parser/token.rs

```diff
@@ -51,6 +51,25 @@ pub enum HexEncodedValue {
     Word([Felt; 4]),
 }
 
+// BINARY ENCODED VALUE
+// ================================================================================================
+
+/// Represents one of the various types of values that have a hex-encoded representation in Miden
+/// Assembly source files.
+#[derive(Debug, Copy, Clone, PartialEq, Eq)]
+pub enum BinEncodedValue {
+    /// A tiny value
+    U8(u8),
+    /// A small value
+    U16(u16),
+    /// A u32 constant, typically represents a memory address
+    U32(u32),
+    /// A single field element, 8 bytes, encoded as 64 binary digits
+    Felt(Felt),
+    // /// A set of 4 field elements, 32 bytes, encoded as a contiguous string of 256 binary digits
+    // Word([Felt; 4]),
```

Comment:
We can probably remove the comment for `Word`. Also, do we need the `Felt` variant?

Reply:
I would remove `Felt` for now as well. If we need it in the future, we can add it then.